File: benchmarks/_foreign/iter.py
import time

def myrange(high):
    current = 0
    while current < high:
        yield current
        current += 1

def fib(n):
    a = 0.0
    b = 1.0
    for i in range(n):
        (a, b) = (b, a + b)
    return a

for outer in range(15):
    start_time = time.time()

    fib(10000000)

    end_time = time.time()

    print("10m range() iterations in", (end_time - start_time) * 1000, "ms")

File: benchmarks/_foreign/iter.js

function* range(max) {
    for(let i = 0; i < max; ++i) {
        yield i;
    }
}

for(let outer of range(15)) {
    var start_time = Date.now()

    for(let i of range(10000000)) {
    }

    var end_time = Date.now()
    var duration_ms = end_time - start_time
    console.log("10m range() iterations took", duration_ms, "ms")
}
